stmt/stmts/insert.go

```diff
@@ -232,6 +232,12 @@ func (s *InsertIntoStmt) Exec(ctx context.Context) (_ rset.Recordset, err error)
 	}
 
 	insertValueCount := len(s.Lists[0])
+	toUpdateColumns, err0 := checkUpdateColumns(s.OnDuplicate, nil, t)
```

Comment:
matchAssignmentToColumn is better than checkUpdateColumns 
I see the result is a map of column index to assignment.